packages/@aws-cdk/aws-ec2-alpha/README.md

```diff
@@ -366,3 +362,104 @@ myVpc.addInternetGateway({
   ipv4Destination: '192.168.0.0/16',
 });
 ```
+
+## Importing an existing VPC
+
+You can import an existing VPC and its subnets using the `VpcV2.fromVpcV2Attributes()` method or an individual subnet using `SubnetV2.fromSubnetV2Attributes()` method.
+
+### Importing a VPC
+
+To import an existing VPC, use the `VpcV2.fromVpcV2Attributes()` method. You'll need to provide the VPC ID, primary CIDR block, and information about the subnets. You can import secondary address as well created through IPAM, BYOIP(IPv4) or enabled through Amazon Provided IPv6. You must provide VPC Id and its primary CIDR block for importing it.
+
+If you wish to add a new subnet to imported VPC, new subnet's IP range(IPv4) will be validated against provided secondary and primary address block to confirm that it is within the the range of VPC.
+
+Here's an example of how to import a VPC with multiple CIDR blocks, IPv6 support, and different subnet types:
+
+In this example, we're importing a VPC with:
+
+ - A primary CIDR block (10.1.0.0/16)
+ - One secondary IPv4 CIDR block (10.2.0.0/16)
+ - Two secondary address using IPAM pool (IPv4 and IPv6)
+ - VPC has Amazon-provided IPv6 CIDR enabled
+ - An isolated subnet in us-west-2a
+ - A public subnet in us-west-2b
+
+```ts
+
+const stack = new Stack();
+
+const importedVpc = VpcV2.fromVpcV2Attributes(this, 'ImportedVPC', {
+  vpcId: 'vpc-XXX',
+  vpcCidrBlock: '10.1.0.0/16',
+  secondaryCidrBlocks: [
+    {
+      cidrBlock: '10.2.0.0/16',
+      cidrBlockName: 'ImportedBlock1',
+    },
+    {
+      ipv6IpamPoolId: 'ipam-pool-XXX',
+      ipv6NetmaskLength: 52,
+      cidrBlockName: 'ImportedIpamIpv6',
+    },
+    {
+      ipv4IpamPoolId: 'ipam-pool-XXX',
+      ipv4IpamProvisionedCidrs: ['10.2.0.0/16'],
+      cidrBlockName: 'ImportedIpamIpv4',
+    },
+    {
+      amazonProvidedIpv6CidrBlock: true,
+    }
+  ],
+  isolatedSubnets: [{
+    subnetName: 'IsolatedSubnet2',
+    subnetId: 'subnet-03cd773c0fe08ed26',
+    subnetType: SubnetType.PRIVATE_ISOLATED,
+    availabilityZone: 'us-west-2a',
+    ipv4CidrBlock: '10.2.0.0/24',
+    routeTableId: 'rtb-0871c310f98da2cbb',
+  }],
+  publicSubnets: [{
+    subnetId: 'subnet-0fa477e01db27d820',
+    subnetType: SubnetType.PUBLIC,
+    availabilityZone: 'us-west-2b',
+    ipv4CidrBlock: '10.3.0.0/24',
+    routeTableId: 'rtb-014f3043098fe4b96',
+  }],
+});
+
+// You can now use the imported VPC in your stack
+
+// Adding a new subnet to the imported VPC
+const importedSubnet = new SubnetV2(this, 'NewSubnet', {
```

Comment:
Like VPC, do you think if there is a need to support importing subnet from another region, or account

Reply:
only possible case I can think of is if some new route added to the imported subnet's route table, but that is not support even for non-imported route tables, meaning we cannot do something like subnet.routetable.addroute(), so not sure if we really need it.